sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1956,4 +1956,30 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     assert(!nonDeterministicQueryPlan.deterministic)
   }
 
+  test("SPARK-38132: Avoid optimizing Not IN subquery") {
```

Comment:
Shall we give more meaningful name for this test case?
For me, the test case looks like checking the correctness of the queries which is irrelevant to avoid `optimizing` something.

Reply:
Thanks, renamed